### PR TITLE
MLIBZ-1525: Expired Session Crash

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR verifies that we are able to correctly refresh MIC sessions when they have expired.

#### Changes
- Don't retry failed refresh requests when we are in the middle of refreshing a MIC session. [af4ce5f](https://github.com/Kinvey/js-sdk/commit/af4ce5fef3495dff74d418e94b356be60211b73c)
- Add unit tests to verify we can refresh MIC sessions. [9949d11](https://github.com/Kinvey/js-sdk/commit/9949d11855101f10dc99820f9b29cb662b1e68b0).

### CODE REVIEW

File: src/request/src/network.js [hunk 1]
```diff
@@ -425,7 +425,7 @@ export class KinveyRequest extends NetworkRequest {
                 properties: this.properties
               });
 
-              return refreshMICRequest.execute()
+              return refreshMICRequest.execute(false, false)
```
Comment (tejasranade) [inline src/request/src/network.js]: @thomasconner What are the two parameters here?

File: src/request/src/response.js [hunk 2]
```diff
@@ -158,9 +158,9 @@ export class KinveyResponse extends Response {
       return super.error;
     }
 
-    if (isDefined(name)) {
-      error.name = name;
-    }
+    // if (isDefined(name)) {
```
Comment (thomasconner) [inline src/request/src/response.js]: It is deliberate. I wanted to make sure it was ok before I removed the code. Maybe we should discuss in standup.

File: src/request/src/network.js (hunk 1 above)
Comment (thomasconner) [inline src/request/src/network.js]: The first parameter is used to return the raw response rather then just the data from the response. The second parameter is used to retry the request in the event a `401` status code is encountered.

Comment (thomasconner) [inline src/request/src/network.js]: See https://github.com/Kinvey/js-sdk/blob/9949d11855101f10dc99820f9b29cb662b1e68b0/src/request/src/network.js#L364

Comment (tejasranade) [inline src/request/src/network.js]: Cool. Makes sense. Do we have any requests that default to `retry = true`?

**Pull request merged**